okhttp/src/main/java/okhttp3/HttpUrl.java

```diff
@@ -772,6 +772,23 @@ public Builder addPathSegment(String pathSegment) {
       return this;
     }
 
+    public Builder addPathSegments(String groupedPathSegments) {
+      if (groupedPathSegments == null) {
+        throw new IllegalArgumentException("groupedPathSegments == null");
+      }
+      int startIndex = 0;
+      int endIndex = groupedPathSegments.indexOf('/');
+      while (endIndex != -1) {
+        String pathSegment = groupedPathSegments.substring(startIndex, endIndex);
```

Comment:
Oooh, you can avoid the substring call. Push will do that for you automatically.